pkg/app/api/grpcapi/api.go

```diff
@@ -188,6 +190,58 @@ func (a *API) ListApplications(ctx context.Context, req *apiservice.ListApplicat
 			Value:    req.Disabled,
 		},
 	}
+
+	if req.EnvId != "" {
+		filters = append(filters, datastore.ListFilter{
+			Field:    "Id",
+			Operator: "==",
+			Value:    req.EnvId,
+		})
+	}
+	// env-id has higher priority than env-name, only use env-name filter in case env-id is not set.
```

Comment:
This function looks env-name is more prioritized than env-id to me.

Reply:
Nop, I saw wrong.

Reply:
maybe the problem is on my comment 😅 What I mean is: the below code block only runs on no env-id passed via command but env-name. In case of env-id passed via command flag, only the above code block is run.

Reply:
Okay, I got it. Then how about using if-else chain as:

```go
if req.EnvId !=  "" {
  ...
} else if req.EnvName != "" {
```

Reply:
To be honest, I'm not a fan of if-else chain, especially when the else condition contains other variables than the first if block 😅 Should I improve by rewriting the comment here?

Reply:
Ok, I got your philosophy and no objection to it. Let's just make the comment clear!

Reply:
thanks, how about "Use env-name as listApplications filter only in case env-id is not set" 👀

Reply:
Looks good

Reply:
🙆‍♀️